clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -43,8 +43,7 @@ const char *SYCL::Linker::constructLLVMSpirvCommand(
     CmdArgs.push_back("-spirv-ext=+all");
     CmdArgs.push_back("-spirv-debug-info-version=legacy");
     CmdArgs.push_back("-spirv-allow-extra-diexpressions");
-    if (C.getArgs().hasArg(options::OPT_fsycl_esimd))
-      CmdArgs.push_back("-spirv-allow-unknown-intrinsics");
+    CmdArgs.push_back("-spirv-allow-unknown-intrinsics=llvm.genx.");
```

Comment:
Definitely not very efficient having to make 2 changes.  This needs to be refactored (not by you, but an issue should be created)

Reply:
Opened https://github.com/intel/llvm/issues/3451.